contracts/Account.sol

```diff
@@ -67,6 +66,44 @@ contract Account is IAccount, Ownable, SafetyWithdraw {
         transferOwnership(governance);
     }
 
+    /**
+     * @notice Adjust the max leverage as insurance pool slides from 100% of target to 0% of target
+     */
+    function realMaxLeverage(address market) public view override returns(int256) {
+        ITracerPerpetualSwaps _tracer = ITracerPerpetualSwaps(market);
+        IInsurance insurance = IInsurance(insuranceContract);
+        int256 baseMaxLeverage = _tracer.maxLeverage();
+
+        int256 percentFull =
+            insurance.getPoolHoldings(market).toInt256()
+            .mul(PERCENT_PRECISION)
+            .div(insurance.getPoolTarget(market).toInt256()).mul(100);
+        int256 cliff = _tracer.INSURANCE_DELEVERAGING_CLIFF();
+
+        if (percentFull > cliff.mul(PERCENT_PRECISION)) {
+            return baseMaxLeverage;
+        }
+
+        // Use int256 for compatibility with baseMaxLeverage
+        //int256 oneXLeverage = 10000;
+        // Linear function intercepting points (1, 0) and (INSURANCE_DELEVERAGING_CLIFF, baseMaxLeverage)
+        // y = mx + b,
+        // where m = INSURANCE_DELEVERAGING_CLIFF - 1
+        //       x = percentFull
+        //       b = 1 (since 1 is the y-intercept)
+        // 1*10,000
+
+        int256 one = PERCENT_PRECISION;
+        int256 realMaxLeverage =
```

Comment:
I'm struggling to understand this maths here. So we take the ratio of baseMaxLeverage to the cliff (eg baseMaxLeverage = 100x, cliff = 10%). We then use this ratio and multiply by how full the pool is to produce the realMaxLeverage?

Reply:
It's just a linear function where the x axis is how full the ins pool is, y axis is realMaxLeverage.

At 0% ins pool, realMaxLeverage is 1, so it passes through point (0, 1).
At INSURANCE_DELEVERAGING_CLIFF% ins pool, realMaxLeverage = baseMaxLeverage, so it passes through point (INSURANCE_DELEVERAGING_CLIFF, baseMaxLeverage).

In between, it scales linearly so I derived a generalised linear function where you can calculat the gradient by using the (x2 - x1)/(y2 - y1) formula, and b will always be 1 because that's the y intercept.

I have modified the comment to fix errors and explain this better.